packages/rocketchat-message-star/client/actionButton.js

```diff
@@ -18,7 +18,10 @@ Meteor.startup(function() {
 			if (RocketChat.models.Subscriptions.findOne({ rid: message.rid }) == null) {
 				return false;
 			}
-			return RocketChat.settings.get('Message_AllowStarring') && !message.starred;
+
+			const starred = _.findWhere(message.starred, {_id: Meteor.userId()});
+
+			return RocketChat.settings.get('Message_AllowStarring') && !starred;
```

Comment:
there is no need to to a `findWhere` if `RocketChat.settings.get('Message_AllowStarring')` isn't `true`..